File: 09_주정윤/week4/10828.py
# 명령의 수
N = int(input())

list=[]
num_list=[]
ans_list=[]

for i in range(N):
    list = input().split()

    if list[0] == "push": #명령어가 push일때
        num_list.append(int(list[1]))
    elif list[0] == "top": #명령어가 top일때
        if len(num_list) == 0:
            ans_list.append(-1)
        else:
            ans_list.append(num_list[-1])
        
    elif list[0] == "size": #명령어가 size일때
        ans_list.append(len(num_list))
    elif list[0] == "empty":
        if len(num_list) == 0:
            ans_list.append(1)
        else:
            ans_list.append(0)
    elif list[0] == "pop":
        if len(num_list) == 0:
            ans_list.append(-1)
        else:
            ans_list.append(num_list.pop())


for i in range(len(ans_list)):
    print(ans_list[i])
    

File: 05_이원영/week4/10828.js
//17-1 10828
const fs = require('fs');
//const input = fs.readFileSync('/dev/stdin').toString().split('\n'); //임시 할당

var num = 7;
var command = ['pop', 'pop', 'push 123', 'top', 'pop', 'top', 'pop'];
var stack = [];

for (let i = 1; i <= num; i++) {
    if (command[i] === 'top') {    //top
        if (stack.length === 0) {
            console.log(-1);
        }

        else {
            console.log(stack[stack.length - 1]);
        }
    }

    else if (command[i] === 'size') {  // size
        console.log(stack.length);
    }

    else if (command[i] === 'empty') {    //empty
        if (stack.length === 0) {
            console.log(1);
        }

        else {
            console.log(0);
        }
    } 
    
    else if (command[i] === 'pop') {   //pop
        if (stack.length === 0) {
            console.log(-1);
        }

        else {
            console.log(stack.pop());
        }
    }

    else {  //push
        stack.push(Number(command[i].split(' ')[1]));
    }
}
